extension/extensionauth/client.go

```diff
@@ -12,27 +12,34 @@ import (
 	"go.opentelemetry.io/collector/extension"
 )
 
-// Client is an Extension that can be used as an authenticator for the configauth.Authentication option.
+// HTTPClient is an Extension that can be used as an HTTP authenticator for the configauth.Authentication option.
 // Authenticators are then included as part of OpenTelemetry Collector builds and can be referenced by their
 // names from the Authentication configuration.
-type Client interface {
+type HTTPClient interface {
 	extension.Extension
 
 	// RoundTripper returns a RoundTripper that can be used to authenticate HTTP requests.
 	RoundTripper(base http.RoundTripper) (http.RoundTripper, error)
+}
+
+// GRPCClient is an Extension that can be used as an HTTP authenticator for the configauth.Authentication option.
+// Authenticators are then included as part of OpenTelemetry Collector builds and can be referenced by their
+// names from the Authentication configuration.
+type GRPCClient interface {
+	extension.Extension
```

Comment:
I don't think we should follow this model, but instead the optional interface model, which means that we will only define the  `PerRPCCredentials()` func.

If we do this, implementations can pick to implement both http and gRPC or just one of them. Also, allows implementations to implement also other types of extensions.